src/NServiceBus.Transport.SqlServer/SqlServerTransportInfrastructure.cs

```diff
@@ -21,6 +27,18 @@ internal SqlServerTransportInfrastructure(SqlServerTransport transport, HostSett
 
             tableBasedQueueCache = new TableBasedQueueCache(addressTranslator, !isEncrypted);
             connectionFactory = CreateConnectionFactory();
+
+            if (typeof(SqlConnection).Namespace != "Microsoft.Data.SqlClient")
+            {
+                return;
+            }
+
+            var informationalVersion = typeof(SqlConnection).Assembly.GetCustomAttributes().OfType<AssemblyInformationalVersionAttribute>().Single();
+            var currentClientVersion = new Version(informationalVersion.InformationalVersion.Split('+').First());
+            if (currentClientVersion < new Version(5, 2, 0))
+            {
+                _logger.WarnFormat("You are using an outdated version '{0}' of Microsoft.Data.SqlClient. We recommend using version 5.2.0 or later by taking an explicit package dependency to avoid issues with memory leaks or transactions running on .NET in the driver.", currentClientVersion);
```

Comment:
This might work since we link to the 5.x breaking changes then

```suggestion
                _logger.WarnFormat("You are using an outdated version '{0}' of Microsoft.Data.SqlClient. We recommend using version 5.2.0 or later by adding a top-level package reference to avoid known problems in older versions of the client. See https://github.com/dotnet/SqlClient/blob/main/release-notes/5.0/5.0.0.md for details on connection string breaking changes before ugprading.", currentClientVersion);
```

Reply:
Where in that link is the relevant connection string info? Seems like we should be pointing directly at that vs. some general release notes.

Reply:
That's the problem. It is only available here https://github.com/dotnet/SqlClient/blob/main/release-notes/4.0/4.0.0.md#encrypt-default-value-set-to-true but then again we talk about the 5.x versions in the text but link to the 4.x version which is super confusing. Then I'd rather point to the general release notes of the 5.x client or the breaking changes section within those release notes.

https://github.com/dotnet/SqlClient/blob/main/release-notes/5.0/5.0.0.md#breaking-changes

Reply:
> the text but link to the 4.x version which is super confusing.

I don't agree. If we need to give them information about the connection string, then a link to the 5.0 breaking changes doesn't accomplish anything.